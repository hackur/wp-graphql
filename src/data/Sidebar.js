import { toGlobalId } from 'graphql-relay';

const path = process.env.WP_SIDEBARS_ENDPOINT || null;
if (!path) {
  throw Error(
    'This endpoint does not exist in WordPress yet. ' +
      'You must install the WordPres GraphQL Middleware plugin.'
  );
}

class Sidebar {
  getID() {
    return toGlobalId(this.constructor.name, this.id);
  }

  static getEndpoint() {
    return path;
  }
}

export default Sidebar;
