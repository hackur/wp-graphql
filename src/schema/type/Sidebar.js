import { GraphQLObjectType, GraphQLList, GraphQLString } from 'graphql';
import WidgetType from 'type/Widget';
import { globalIdField, name } from 'field/identifier';
import description from 'field/description';
import { registerNodeType, NodeInterface } from 'type/relayNode';

const SidebarType = new GraphQLObjectType({
  name: 'Sidebar',
  description: 'A widget area.',
  interfaces: [NodeInterface],
  fields: {
    id: globalIdField(),
    ...name,
    ...description,
    classname: {
      type: GraphQLString,
      description: 'Extra CSS class to assign to the sidebar in the Widgets interface.',
      resolve: sidebar => sidebar.class,
    },
    before_widget: {
      type: GraphQLString,
      description:
        "HTML content to prepend to each widget's HTML output when assigned to this sidebar.",
    },
    after_widget: {
      type: GraphQLString,
      description:
        "HTML content to append to each widget's HTML output when assigned to this sidebar.",
    },
    before_title: {
      type: GraphQLString,
      description: 'HTML content to prepend to the sidebar title when displayed.',
    },
    after_title: {
      type: GraphQLString,
      description: 'HTML content to append to the sidebar title when displayed.',
    },
    widgets: {
      type: new GraphQLList(WidgetType),
      description: 'HTML widgets associated with the sidebar.',
    },
  },
});

registerNodeType(SidebarType);

export default SidebarType;
