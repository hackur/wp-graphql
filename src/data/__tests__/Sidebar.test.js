import Sidebar from 'data/Sidebar';

describe('Test Sidebar data access', () => {
  test('Get endpoint', () => {
    expect(Sidebar.getEndpoint()).toMatchSnapshot();
  });
});
