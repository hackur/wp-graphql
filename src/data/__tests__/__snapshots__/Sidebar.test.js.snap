// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Sidebar data access Get endpoint 1`] = `"graphql/v1/sidebars"`;
